content/posts/2019-08-03---Going-from-ember-to-vue.md

```diff
@@ -0,0 +1,284 @@
+But I found it was hard to jump straight from Ember into React; not impossible, but I felt I had to unlearn some habits that I had formed.
+For the past few months though, I have been working on a Vue project and I have found the leap from Ember to Vue was much more straight forward.
+
+So, I was going to take some time and write about the similarities and differences I have experienced.
+Even if you still love Ember and plan on working with it for a while, I would encourage you to look at Vue just to see what else is out there.
+
+# Some big differences
+
+Much like React, Vue is a component library -- which means you'll be using components for <strong>everything.</strong>
+That means no controllers or route objects.
+When you create a route, you are pointing to a component.
+There is literally nothing special about this component -- it can be literally any component that you want (in fact, you can use it both as a route _and_ component on the page... although, you really shouldn't).
+
+In my opinion, this is a good habit to get into as it will help you transition to something like React down the line if you have to.
+
+## Batteries aren't included
+
+![Batteries](https://i.imgur.com/65SKMnM.jpg?1)
+<aside>Photo by <a href="https://www.pexels.com/@hilaryh">Hilary Halliwell on Pexels</a></aside>
+
+This isn't meant to be a knock on Vue, but it doesn't come with as much stuff out of the box as Ember.
+In fact, this is probably a huge plus.
+[Vue CLI](https://cli.vuejs.org/) allows you to generate a project quickly, but much like Create React App, you are getting a pretty simple UI application with webpack pre-configured for you.
+And yes, webpack over Brocolli is another big difference... but it's probably a good one.
+
+However, anything you want to do in ember, probably already has an equivalent library in Vue.
+And most of these libraries are maintained by core team members, so they feel very official.
+With just a couple more `vue add` commands, you can get routing and state management in your app, so it's really not a _huge_ knock against Vue.
+
+## Components are one file
+
+I found this weird at first, but boy, do I like components with both markup and JavaScript in the same file.
+I hated having to bounce between files so often in Ember (especially when the community wanted to move away from pod structure).
+
+<figure>
+  <blockquote>
+    <p>Ugh, markup and JS in the same file?? I believe in separation of concerns.</p>
+    <footer>
+      <cite>— People who want you to know they're smart.</cite>
+    </footer>
+  </blockquote>
+</figure>
+
+I've actually heard this complaint a few times about "I don't like JS/HTML/CSS in the same file because I like separation of concerns."
+I get the sentiment, but to be clear, just because "concerns" are in the same file, doesn't mean they aren't separated.
+The JavaScript of a component is too tightly tied to the markup (as it should be), so why force yourself to bounce between the two files?
```

Comment:
the pattern is actually handy in react -- where you have a "smart" component (js), and a "dump component" (template). It's handy when your component does a lot of things